force-app/main/default/classes/Grid.cls

```diff
@@ -0,0 +1,124 @@
+/**
+* @description tracks the position of a user in a 5x5 grid as x and y coordinates. 
+* The coordinates are index origin 0. The default position is the top left,
+* but the user can specify the position in the constructor.
+*/
+public with sharing class Grid 
+{
+    // holds the x coordinate within the 5x5 grid
+    public Integer x
+    {
+        get { return x; }
+        set 
+        { 
+            if (value < 0) 
+            {
+                x = 0;
+            } 
+            else if (value > 4)
+            {
+                x = 4;
+            }
+            else
+            {
+                x = value;
+            }
+        }
+    }
+
+    // holds the y coordinate within the 5x5 grid
+    public Integer y
+    {
+        get { return y; }
+        set 
+        { 
+            if (value < 0) 
```

Comment:
As the checks in both setters are the same, you could offload to another method. That way if the grid dimensions change, you only need to update in one place.

Reply:
good point, wasn't too sure if the method that the setter calls is best to be an instance or static method? and if there is a good place to put this kind of 'utility' method in the class (at the end)?